pkg/kudoctl/util/repo/bundle.go

```diff
@@ -0,0 +1,165 @@
+package repo
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+// This is an abstraction which abstracts the underlying bundle, which is likely file system or compressed file.
+// There should be a complete separation between retrieving a bundle if not local and working with a bundle.
+
+// Bundle is an abstraction of the collection of files that makes up a package.  It is anything we can retrieve the PackageCRDs from.
+type Bundle interface {
```

Comment:
` Bundle is an abstraction of the collection of files that makes up a package.` ? so why is it not package? collection of files that make up package == package :)